azure-batch/azure/batch/custom/patch.py

```diff
@@ -0,0 +1,287 @@
+        added.
+    :type job_id: str
+    :param value: The collection of tasks to add.
+    :type value: list of :class:`TaskAddParameter
+        <azure.batch.models.TaskAddParameter>`
+    :param task_add_collection_options: Additional parameters for the
+        operation
+    :type task_add_collection_options: :class:`TaskAddCollectionOptions
+        <azure.batch.models.TaskAddCollectionOptions>`
+    :param dict custom_headers: headers that will be added to the request
+    :param bool raw: returns the direct response alongside the
+        deserialized response
+    """
+
+    def __init__(
+            self,
+            client,
+            job_id,
+            tasks_to_add,
+            task_add_collection_options=None,
+            custom_headers=None,
+            raw=False,
+            **kwargs):
+        # No complex operations - No lock needed
+        self._has_early_termination_error = False
+
+        # Append operations thread safe
+        # Only read once all threads have completed
+        self._failures = collections.deque()
+
+        # synchronized through lock variables
+        self.error = None  # Only written once all threads have completed
+        self._max_tasks_per_request = MAX_TASKS_PER_REQUEST
+        self._tasks_to_add = collections.deque(tasks_to_add)
+
+        self._error_lock = threading.Lock()
+        self._max_tasks_lock = threading.Lock()
+        self._pending_queue_lock = threading.Lock()
+
+        # Variables to be used for task add_collection requests
+        self._client = TaskOperations(
+            client._client, client.config, client._serialize, client._deserialize)
+        self._job_id = job_id
+        self._task_add_collection_options = task_add_collection_options
+        self._custom_headers = custom_headers
+        self._raw = raw
+        self._kwargs = dict(**kwargs)
+
+    def _bulk_add_tasks(self, results_queue, chunk_tasks_to_add):
+        """Adds a chunk of tasks to the job and retry if chunk
+
+        :param results_queue: Queue to place the return value of the request
+        :type results_queue: collections.deque
+        :param chunk_tasks_to_add: Chunk of at most 100 tasks with retry details
+        :type chunk_tasks_to_add: list[~TrackedCloudTask]
+        """
+
+        try:
+            add_collection_response = self._client.add_collection(
+                self._job_id,
+                chunk_tasks_to_add,
+                self._task_add_collection_options,
+                self._custom_headers,
+                self._raw)
+        except BatchErrorException as e:
+            # In case of a chunk exceeding the MaxMessageSize split chunk in half
+            # and resubmit smaller chunk requests
+            # TODO: Replace string with constant variable once available in SDK
+            if e.error.code == "RequestBodyTooLarge":  # pylint: disable=no-member
+                # In this case the task is misbehaved and will not be able to be added due to:
```

Comment:
This is effectively the same content we had in the extension, right?

Reply:
It looks like it to me

Reply:
Yea only changes were python stylistic stuff Laurent recommended, and a couple adjustments to the objects paths due to being in the python sdk project instead of extensions. 

Only 'new' code is the patch client stuff